core-tests/src/test/java/org/mule/runtime/core/internal/processor/simple/ParseTemplateProcessorTestCase.java

```diff
@@ -81,7 +92,7 @@ public void testParseTemplateWithBothLocationAndContentDefined() throws Initiali
     parseTemplateProcessor.initialise();
   }
 
-  @Test(expected = IllegalArgumentException.class)
+  @Test(expected = InitialisationException.class)
```

Comment:
Use the rule instead. It will allow you to match the error message as well to make sure the correct failure is occurring.